src/charactersheet/components/column-navigation/index.js

```diff
@@ -0,0 +1,190 @@
+import autoBind from 'auto-bind';
+import ko from 'knockout';
+import template from './index.html';
+import './index.css';
+
+export class NavigationComponentViewModel {
+
+    sizes = ['xs', 'sm', 'md', 'lg'];
+
+    constructor(params) {
+        autoBind(this);
+        this.collapseOn = params.collapseOn || 'xs';
+        this.columns = ko.observableArray([]);
+        this.estimatedColumns = params.estimatedColumns || 3;
+        this.endColumnPadding = params.endColumnPadding || 3;
+
+        this.push({ nodes: params.templateNodes, context: params.context });
+    }
+
+    // UI
+
+    nextSizeUp = ko.pureComputed(() => {
+        const sizeIndex = this.sizes.indexOf(ko.unwrap(this.collapseOn));
+        const nextSizeIndex = sizeIndex + 1;
+        if (nextSizeIndex < this.sizes.length) {
+            return this.sizes[nextSizeIndex];
+        } else {
+            return null;
+        }
+    });
+
+    existingColumnWidth = ko.pureComputed(() => (
```

Comment:
can you explain/add comments/docs as to what this is calculating and what the magic numbers here are?

Reply:
This component uses Bootstrap's column widths, so 12 is the full width `col-xs-12` etc.